[ncurses] Initial plan and tests

MindNumbing opened the pull request:
Signed-off-by: Steven Marshall <SMarshall@chef.io>

Commit:
[ncurses] Initial plan and tests

Signed-off-by: Steven Marshall <SMarshall@chef.io>

gavindidrichsen commented on controls/ncurses_works.rb:
```diff
@@ -0,0 +1,176 @@
+  hab_pkg_path = command("hab pkg path #{plan_ident}")
+  describe hab_pkg_path do
+    its('exit_status') { should eq 0 }
+    its('stdout') { should_not be_empty }
```
Extra stderr check please:  its('stderr') { should be_empty }

gavindidrichsen reviewed (changes requested):
Good work here; lots of binary checks.  One small change to update.

Commit:
[ncurses] Various refactors to existing code

Added inspec.lock to .gitignore
Updated README.md to latest binary format
Added plan_name to attributes.yml
Updates controls to add more description, added version checking
Updated inspec.yml to latest version

Signed-off-by: Steven Marshall <SMarshall@chef.io>

gavindidrichsen reviewed (approved):
LGTM
